File: Luogu/Greedy/P4995.py
n_strs = input().split(' ')
n = int(n_strs[0])

in_nums = []
in_nums_strs = input().split(' ')
for str in in_nums_strs:
    if ('' != str):
        in_nums.append(int(str))

in_nums.sort()

stones = []
for i in range(n):
    stones.append(in_nums[n - 1 - i])
    stones.append(in_nums[i])

sum = stones[0] * stones[0]
for i in range(1, n):
    sum += (stones[i] - stones[i - 1]) * (stones[i] - stones[i - 1])

print(sum)

File: Luogu/Greedy/P4995.cc
#include <iostream>
#include <vector>
#include <algorithm>

int main(void) {
    int n = 0;
    long long sum = 0;

    std::cin >> n;

    std::vector<int> input(n, 0);
    std::vector<int> stones(2 * n, 0);
    for (int i = 0; i < n; i ++) {
        std::cin >> input[i];
    }

    std::sort(input.begin(), input.end());

    for (int i = 0, j = n - 1; i < n && j >= 0; i ++, j --) {
        stones[2 * i] = input[j];
        stones[2 * i + 1] = input[i];
    }

    sum += (long long)stones[0] * stones[0];
    for (int i = 1; i < n; i ++) {
        sum += (long long)((stones[i - 1] - stones[i]) *
                (stones[i - 1] - stones[i]));
    }

    std::cout << sum << std::endl;

    return 0;
}